src/pages/index.js

```diff
@@ -4,114 +4,247 @@ import FormValidator from "../components/FormValidator.js";
-  const cardData = {
-    name: inputValue.title,
-    link: inputValue.description,
-  };
-  cardSection.addItem(createCard(cardData));
-  addCardForm.reset();
-  addCardFormValidator.disableSubmitButton();
-  addCardPopup.close();
+function handleDeleteClick(card) {
+  confirmDeletePopup.open();
+  confirmDeletePopup.setSubmitAction(() => {
+    confirmDeletePopup.setDeleteLoading(true);
+    api
+      .deleteCard(card._id)
+      .then((result) => {
+        card.handleDeleteCard(result);
+        confirmDeletePopup.close();
+      })
+      .catch((err) => {
+        alert(`${err} Failed to delete post.`);
+      })
+      .finally(() => {
+        confirmDeletePopup.setDeleteLoading(false);
+      });
+  });
 }
 
-function handleImageClick(data) {
-  imagePreviewPopup.open(data);
+/* -------------------------------------------------------------------------- */
+/*                                  Functions                                 */
+/* -------------------------------------------------------------------------- */
+
+function handleImageClick(card) {
+  previewImagePopup.open(card);
 }
 
-function createCard(data) {
-  const card = new Card(data, "#card-template", handleImageClick);
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleAddLike
+  );
   return card.getView();
 }
 
-function renderCard(data, placement = "prepend") {
-  const cardElement = createCard(data);
-  cardListEl[placement](cardElement);
-}
+/* -------------------------------------------------------------------------- */
+/*                               Event Listeners                              */
+/* -------------------------------------------------------------------------- */
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: (data) => {
-      //get the card html element
-      const cardElement = createCard(data);
-      //stick the card element on the page
-      cardSection.addItem(cardElement, "append");
-    },
-  },
-  ".cards__list"
-);
-cardSection.renderItems();
+avatarImage.addEventListener("click", () => {
+  avatarEditForm.open();
+});
+
+pencil.addEventListener("click", () => {
+  avatarEditForm.open();
+});
 
-/* Event Listeners */
+addNewCardBtn.addEventListener("click", () => {
+  addCardFormValidator.resetValidation();
+  addImageForm.open();
+});
 
-profileEditButton.addEventListener("click", () => {
-  const { name, description } = userInfo.getUserInfo();
+profileEditBtn.addEventListener("click", () => {
+  const { name, description } = profileUserInfo.getUserInfo();
   profileTitleInput.value = name;
-  profileDescriptionInput.value = description;
-  editCardFormValidator.resetValidation();
-  profileEditPopup.open();
+  profileSubtitleInput.value = description;
+  profileEditForm.open();
+});
+
+modalImage.addEventListener("click", () => {
+  previewImagePopup.open();
```

Comment:
* When I click on the `image modal`, there are errors https://snipboard.io/BUQxTe.jpg 


Please, delete it 

It's not needed for the image modal. You open the modal only when you click on the card images